### DISCUSSION TITLE
How to get data in onModelAfterCreate (JavaScript)?

### DISCUSSION THREAD

**Discussion by nicetomytyuk** [pocketbase/pocketbase#3393] [Q&A] [answered]
I need to capture che created data onModelAfterCreate and update another table based on the data that has been created in the hook.

The issue is that when i serialize the e.model, I'm able to see all the fields I need, but when I try to parse the value, I always get undefined.

For example:

```
onModelAfterCreate((e: any) => {
    try {
        console.log(JSON.stringify(e.model))
        console.log(e.model.group)
    } catch (e) {
        console.log(e)
    }
}, "queue")
```

The first console log returns:

` {"collectionId":"seabb6g0b822o6y","collectionName":"queue","created":"2023-09-27 12:24:53.911Z","group":"mvgu8ctv87116yb","id":"lb2q6mfc714ia56","number":11,"status":"waiting","updated":"2023-09-27 12:24:53.911Z"}`

But the second one returns undefined.

So how can I parse the group value in that hook?

**Comment by ganigeorgiev** [comment]
`e.model` in your case is a `Record` and you have to use the [getter and setters](https://pocketbase.io/docs/js-records/#getset-record-fields), aka. try with `e.model.get("group")`

**Reply by nicetomytyuk** [reply]
The `e` is a `core.ModelEvent` and the `e.model` is a `core.BaseModelEvent.model` so even if I try to set the `e` to any and try to use the .get I will still get null.

![image](https://github.com/pocketbase/pocketbase/assets/30237552/dcadd1b5-16a5-485c-bdd8-384646e62a84)

**Reply by ganigeorgiev** [reply, marked as answer]
`e.model` holds the base `Model` interface which in your case is `Record` (`Record` implements `Model`). You can safely ignore the TS error (_I'll experiment with the next release to see if I can update the generated typings for the `e.model` field_).

Do you have a `queue` field? From your initial example I thought the field name was `group`.

**Reply by nicetomytyuk** [reply]
@ganigeorgiev oh damn it was a "typo" I was thinking about the table and not the field when I was trying the `.get` so yeah, by trying to fetch the field with `.get('group')` works as expected. 

To skip the TS error I just set the `e` type to `any`.

Thank you!
